extensions/skyportal/static/js/components/TNSInfo.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import PropTypes from "prop-types";
+
+import * as tnsInfoActions from "../ducks/tnsInfo";
+
+const TNSInfo = ({ objID }) => {
+  const dispatch = useDispatch();
+  const [requestSubmitted, setRequestSubmitted] = useState(false);
+  const tnsInfo = useSelector((state) => state.tnsInfo);
+
+  useEffect(
+    function fetchTNSInfo() {
+      if (!requestSubmitted) {
+        dispatch(tnsInfoActions.fetchTNSInfo(objID));
+        setRequestSubmitted(true);
+      }
+    },
+    [objID, dispatch, requestSubmitted]
+  );
+
+  if (tnsInfo === null) {
+    return <>Fetching TNS data...</>;
+  }
+  return (
+    <span>
+      {tnsInfo?.name ? (
+        typeof tnsInfo.name === "string" ? (
```

Comment:
Just wanna be extra careful that we're not calling object instance methods that don't exist and crashing the whole source page